os/tools/mksyscall.c

```diff
@@ -246,9 +248,10 @@ static void generate_proxy(int nparms)
 
 	fprintf(stream, "#include <syscall.h>\n\n");
 
-	cond_parm = get_parm(COND_INDEX);
-	if (cond_parm[0] != '\0')
```

Comment:
We need this check for cond_param. 
Please change the if statement as below. 
`if (cond_parm[0] != '\0' && check_special_case() == true)`